### PR TITLE
Compose services

### PR DESCRIPTION
**Pull request by xresende**
This PR is based on PR https://github.com/lnls-sirius/docker-machine-applications/pull/6 (already merged)

### CODE REVIEW

**Comment by fernandohds564** [comment]
@xresende, I think we could also separate the `bo-corrs` and the `dclinks` IOCs by `ia`, just like the `si-corrs`, `si-trims` and `si-skews`. What do you think?

File: services/docker-stack-si-ps.yml [hunk 1]
```diff
@@ -0,0 +1,2472 @@
...
+      replicas: 1
+      restart_policy:
+        condition: any
+    logging:
+      driver: "json-file"
+      options:
+        max-file: "10"
+        max-size: "10m"
+    networks:
+      - ioc-network
```
Comment (anacso17) [inline services/docker-stack-si-ps.yml:115]: it seems this lines are duplicated

File: services/docker-stack-si-ps.yml [hunk 2]
```diff
@@ -0,0 +1,2472 @@
...
+
+    sexts-sfb12-sdb3-sfp12-sdp23:
```
Comment (anacso17) [inline services/docker-stack-si-ps.yml:201]: is the indentation correct here?

File: services/docker-stack-si-ps.yml [hunk 3]
```diff
@@ -0,0 +1,2472 @@
...
+      driver: "json-file"
+      options:
+        max-file: "10"
+        max-size: "10m"
+    networks:
+      - ioc-network
+  
+  sexts-sfa12-sda3p1-sfb0-sdb01:
+    image: dockerregistry.lnls-sirius.com.br/fac/fac-iocs:2020-03-02-test-si-corrs
+    command: bash -c 'source /etc/profile && /ioc-apps/si-ps-si-ps-sexts-sfa12-sda3p1-sfb0-sdb01.bash'
```
Comment (anacso17) [inline services/docker-stack-si-ps.yml:147]: Is the **si-ps-si-ps-** correct?

File: Makefile [hunk 4]
```diff
@@ -37,1046 +36,1158 @@ image-pull-fac-base:
...
+service-start-highstack-si-ps:
 	cd services; docker stack deploy -c docker-stack-si-ps.yml facs-si-ps
 
-service-stop-set1-as-ti-conv:
+service-stop-highstack-as-ti-conv:
 	cd services; docker stack rm facs-as-ti
 
-service-start-set1-as-ti-conv:
+service-start-highstack-as-ti-conv:
 	cd services; docker stack deploy -c docker-stack-as-ti.yml facs-as-ti
```
Comment (anacso17) [inline Makefile]: is **as-ti-conv** correct?

File: Makefile [hunk 5]
```diff
@@ -37,1046 +36,1158 @@ image-pull-fac-base:
...
-
-service-facs-si-ti-trims-skews-start:
-	cd services; docker stack deploy -c docker-stack-si-ti-trims-skews.yml facs-si-ti-trims-skews
-
-service-facs-as-ps-dclinks-tbts-bodip-stop:
+TARGET_HIGHSTACK_DEPS_STOP := service-stop-highstack-as-ps-dclinks
+TARGET_HIGHSTACK_DEPS_STOP += service-stop-highstack-li-ps
+TARGET_HIGHSTACK_DEPS_STOP += service-stop-highstack-tb-ps
+TARGET_HIGHSTACK_DEPS_STOP += service-stop-highstack-bo-ps
+TARGET_HIGHSTACK_DEPS_STOP += service-stop-highstack-si-ps
```
Comment (anacso17) [inline Makefile:154]: missing **ts-ps** target here

Comment (anacso17) [reply Makefile:154]: same in other groups

File: Makefile [hunk 6]
```diff
@@ -37,1046 +36,1158 @@ image-pull-fac-base:
...
+
+
+
+
+
+
+
+
+
+
```
Comment (anacso17) [inline Makefile:260]: are these spaces purposeful?

File: services/docker-stack-si-ps.yml (hunk 1 above)
Comment (xresende) [reply services/docker-stack-si-ps.yml]: fixed.

File: services/docker-stack-si-ps.yml (hunk 2 above)
Comment (xresende) [reply services/docker-stack-si-ps.yml:173]: another duplicate here. the duplicate also has wrong indentation.

File: services/docker-stack-si-ps.yml (hunk 3 above)
Comment (xresende) [reply services/docker-stack-si-ps.yml:119]: no. string replacement error. fixed.

File: Makefile (hunk 6 above)
Comment (xresende) [reply Makefile:260]: no. ms code's 'code collapsing' feature was hiding these lines from me. fixed.

File: Makefile (hunk 5 above)
Comment (xresende) [reply Makefile:154]: fixed

File: Makefile (hunk 4 above)
Comment (xresende) [reply Makefile]: fixed

File: Makefile [hunk 7]
```diff
@@ -37,1046 +36,1158 @@ image-pull-fac-base:
...
+service-start-lowstack-si-ti-trims-skews:
+	cd services; docker stack deploy -c docker-stack-si-ti-trims-skews.yml facs-si-ti-trims-skews
+
+
+
+
+
+
+
+
```
Comment (anacso17) [inline Makefile]: there are also spare spaces here

Comment (xresende) [reply Makefile]: fixed it too.

**Comment by xresende** [comment]
> @xresende, I think we could also separate the `bo-corrs` and the `dclinks` IOCs by `ia`, just like the `si-corrs`, `si-trims` and `si-skews`. What do you think?

@fernandohds564 , i began splitting services according to estimates of cpu usage. [did you see it in bash files?]. BO corrs, groups in sets of five rack rooms, are nowm after PR#6, above average CPU usage/container. i think it is now natural to further split these IOCs.

On the other hand, dclink IOCs, gathered as they are, lead to containers with cpu usage below average. further splitting them does not help much. but maybe it is a good idea to have them split just like all other IOCs controlling pwrsupplies in instrumentation areas, for uniformity...

**Comment by fernandohds564** [comment]
nice, @xresende ! I was thinking more about uniformity indeed. I forgot about CPU usage. Maybe that's a stronger reason in favor of keeping them as they are now...

**Review by fernandohds564 — APPROVED** [review]

**Comment by xresende** [comment]
> nice, @xresende ! I was thinking more about uniformity indeed. I forgot about CPU usage. Maybe that's a stronger reason in favor of keeping them as they are now...

uniformity, ok.

the problem is that we still have fixed feelings about too many containers. we don't exactly know the impact on performance due to overhead in service mgmt. that's why i hesitate to split. but we can go the other way around: start conveniently with more containers and then observe performance/overhead and gather back again, if necessary.

after I merge this PR I will create another splitting corrs, dclinks, and additional makefile targets, ok ?

**Comment by fernandohds564** [comment]
ok @xresende !

**Review by anacso17 — APPROVED** [review]

**Pull request merged**